### PR TITLE
[new] Add Mavo w/ auto-update

### PR DESCRIPTION
**Pull request by DmitrySharabin**
# Pull request template for new library

Pull request for issue: #
Related issue(s): # #

Checklist for **Pull request** or **lib adding request issue** follows the conventions.

Note that if you are using a distribution purpose repository/package, please also provide the url and other related info like popularity of the source code repo/package.

## Profile of the lib

* Git repository (required): https://github.com/mavoweb/mavo
* Official website (optional, not the repository): https://mavo.io/
* NPM package url (optional): https://www.npmjs.com/package/mavo
* License and its reference: MIT
* GitHub / Bitbucket popularity (required):
  * Count of watchers: 68
  * Count of stars: 2295
  * Count of forks: 128
* NPM download stats (optional):
  * Downloads in the last day:
  * Downloads in the last week:
  * Downloads in the last month:

## Essential checklist

* [ ] I'm the author of this library
  * [ ] I would like to add link to the page of this library on CDNJS on website / readme
* [x] This lib was not found on cdnjs repo
* [x] Doesn't already exist / has duplicate issue and PR
* [x] The lib has notable popularity
  * [x] More than 200 [Stars / Watchers / Forks] on [GitHub / Bitbucket]
  * [ ] More than 800 downloads stats per month on npm registry
* [x] Project has public repository on famous online hosting platform (or been hosted on npm)

## Auto-update checklist

* [ ] Has valid tags for each versions (for git auto-update)
* [x] Auto-update setup
* [x] Auto-update target/source is valid.
* [x] Auto-update filemap is correct.

## Git commit checklist

* [ ] The first line of commit message is less then 50 chars; clean, clear and easy to understand.
* [ ] The parent of the commit(s) in the PR is not older than 3 days.
* [ ] Pull request is sent from a non-master branch with a meaningful name.
* [ ] Separate unrelated changes into different commits.
* [ ] Use rebase to squash/fixup dummy/unnecessary commits into only one commit.
* [ ] Close corresponding issue in commit message
* [ ] Mention related issue(s), people in commit message, comment.

### CODE REVIEW

File: mavo/package.json [hunk 1]
```diff
@@ -0,0 +1,57 @@
+{
+  "name": "mavo",
```
Comment (MattIPv4) [inline mavo/package.json]: You need a version defined in this file.

File: mavo/package.json [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+  "description": "Create web applications with HTML and CSS",
+  "main": "dist/mavo.js",
```
Comment (MattIPv4) [inline mavo/package.json]: You need to include all the files for the current version in this PR.

File: mavo/package.json [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+  "scripts": {
```
Comment (MattIPv4) [inline mavo/package.json]: Scripts is not needed in this file.

File: mavo/package.json [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+    "type": "git",
+    "url": "https://github.com/mavoweb/mavo.git"
+  },
+  "author": "Lea Verou",
+  "license": "MIT",
+  "bugs": {
+    "url": "https://github.com/mavoweb/mavo/issues"
+  },
+  "homepage": "https://mavo.io",
+  "browserslist": [
```
Comment (MattIPv4) [inline mavo/package.json]: Browser support is not needed in this file.

File: mavo/package.json [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+    "last 4 versions",
+    "IE 11"
+  ],
+  "devDependencies": {
```
Comment (MattIPv4) [inline mavo/package.json]: Dependencies are not needed in this file.

File: mavo/package.json [hunk 6]
```diff
@@ -0,0 +1,57 @@
+{
```
Comment (MattIPv4) [inline mavo/package.json]: This file needs to live in `cdnjs/ajax/libs/mavo`, not `cdnjs/mavo`.

File: mavo/package.json [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
```
Comment (MattIPv4) [inline mavo/package.json]: This file can be removed.

File: ajax/libs/mavo/package.json [hunk 8]
```diff
@@ -0,0 +1,28 @@
+{
+  "name": "mavo",
+  "version": "0.2.0",
+  "description": "Create web applications with HTML and CSS",
+  "main": "dist/mavo.js",
```
Comment (MattIPv4) [inline ajax/libs/mavo/package.json]: As your base path is `dist`, this just needs to be `mavo.js`

File: ajax/libs/mavo/mavo.js [hunk 9]
```diff
@@ -0,0 +1,11178 @@
...:function(t){t&&t.parentNode.insertBefore(this,t.nextSibling)},start:function(t){t&&t.insertBefore(this,t.firstChild)},around:function(t){t&&t.parentNode&&r.before(this,t),this.appendChild(t)}},r.Array=function(t){this.subject=t},r.Array.prototype={all:function(t){var e=r.$(arguments).slice(1);return this[t].apply(this,e)}},r.add=t(function(t,e,n,i){n=r.extend({$:!0,element:!0,array:!0},n),"function"==r.type(e)&&(!n.element||t in r.Element.prototype&&i||(r.Element.prototype[t]=function(){return this.subject&&r.defined(e.apply(this.subject,arguments),this.subject)}),!n.array||t in r.Array.prototype&&i||(r.Array.prototype[t]=function(){var t=arguments;return this.subject.map(function(n){return n&&r.defined(e.apply(n,t),n)})}),n.$&&(r.sources[t]=r[t]=e,(n.array||n.element)&&(r[t]=function(){var e=[].slice.apply(arguments),i=e.shift(),o=n.array&&Array.isArray(i)?"Array":"Element";return r[o].prototype[t].apply({subject:i},e)})))},0),r.add(r.Array.prototype,{element:!1}),r.add(r.Element.prototype),r.add(r.setProps),r.add(r.classProps,{element:!1,array:!1});var i=document.createElement("_");r.add(r.extend({},HTMLElement.prototype,function(t){return"function"===r.type(i[t])}),null,!0)}();
```
Comment (MattIPv4) [inline ajax/libs/mavo/mavo.js]: The files need to go in the correct version folder. `cdnjs/ajax/libs/mavo/0.2.0/`

File: ajax/libs/mavo/package.json (hunk 8 above)
Comment (DmitrySharabin) [reply ajax/libs/mavo/package.json]: Did I get it right: I should change it to "main": "mavo.js"?

Comment (MattIPv4) [reply ajax/libs/mavo/package.json]: Yup.

Comment (DmitrySharabin) [reply ajax/libs/mavo/package.json]: Thank you! And sorry for all the mistakes I make. 😔 I send you a PR for the first time. I'll do my best next time.

**Comment by DmitrySharabin** [comment]
Agree. I was also thinking about it. Yes, Mavo [published to npm](https://www.npmjs.com/package/mavo/v/0.2.0).
If I add this (instead of the git part) to the `package.json` will it work?
```json
"npmName": "mavo",
  "npmFileMap": [
    {
      "basePath": "dist",
      "files": [
        "**/*"
      ]
    }
  ]
```
I have doubts about the `basePath` key. Should I keep it or remove it?

File: ajax/libs/mavo/package.json [hunk 10]
```diff
@@ -0,0 +1,27 @@
...
+    "url": "https://github.com/mavoweb/mavo/issues"
+  },
+  "homepage": "https://mavo.io",
+  "autoupdate": {
+    "npmName": "mavo",
+    "npmFileMap": [
+      {
+        "basePath": "dist",
+        "files": [
+          "**/*"
```
Comment (MattIPv4) [inline ajax/libs/mavo/package.json]:
```suggestion
          "*.@(js|css)"
```

Looking at the files in the npm dist directory, it looks like you'll only want the JS/CSS files at the root and nothing else?

[Glob test](https://www.digitalocean.com/community/tools/glob?glob=%2A.%40%28js%7Ccss%29&tests=maps%2Fmavo-nodeps.js.map&tests=maps%2Fmavo.es5.min.js.map&tests=maps%2Fmavo.css.map&tests=maps%2Fmavo.min.js.map&tests=maps%2Fmavo.es5.js.map&tests=maps%2Fmavo.js.map&tests=mavo-nodeps.js&tests=0.1.4%2Fmaps%2Fmavo.es5.min.js.map&tests=0.1.4%2Fmaps%2Fmavo.css.map&tests=0.1.4%2Fmaps%2Fmavo.min.js.map&tests=0.1.4%2Fmaps%2Fmavo.es5.js.map&tests=0.1.4%2Fmaps%2Fmavo.js.map&tests=0.1.4%2F.DS_Store&tests=0.1.4%2Fmavo.css&tests=0.1.4%2Fmavo.es5.js&tests=0.1.4%2Fmavo.js&tests=0.1.4%2Fmavo.min.js&tests=0.1.4%2Fmavo.es5.min.js&tests=index.html&tests=.DS_Store&tests=mavo.css&tests=deps.js&tests=mavo.es5.js&tests=mavo.js&tests=mavo.min.js&tests=mavo.es5.min.js)

**Comment by MattIPv4** [comment]
@zackbloom Do we have any status on a new CI (branch protection rules expected the CI to provide a approving review currently as well as a huamn [thus, two required reviews])?

**Comment by zackbloom** [comment]
@xtuc will be back on Monday I believe and should be able to provide an update.

**Comment by DmitrySharabin** [comment]
Hey folks! Sorry for disturbing you. But developers in Russia need your help. 😊
[Netlify](https://www.netlify.com/) (where Mavo is being hosted) and a bunch of other resources are blocked in Russia as a result of the war with the Telegram messenger. So Mavo is not accessible in Russia without using a VPN at the moment. That is why Lea and I decided to host a copy of Mavo at your CDN.
Is there a chance you'll be able to merge this PR in the near future? So we can spread knowledge about this wonderful library far beyond the USA.
Thanks in advance!

**Comment by MattIPv4** [comment]
The current priority is getting the cdnjs website back up and working (#13612). Once that's done, I believe we should be able to shift focus to the CI.

**Comment by DmitrySharabin** [comment]
Got it. Thank you and good luck! :)

**Comment by DmitrySharabin** [comment]
@MattIPv4 Hey there, Matt! Is there a chance you could merge my PR in the near future? I'd be grateful for any information you can give me. Thanks!

**Comment by MattIPv4** [comment]
Unfortunately, I physically do not have the ability to merge it without a second review. We have a meeting with the folks at Cloudflare next week to discuss what the plan is going foward.

**Comment by DmitrySharabin** [comment]
I got it. I continue patiently wait then. Thank you for your answer.

File: ajax/libs/mavo/package.json [hunk 11]
```diff
@@ -0,0 +1,27 @@
...
+    "type": "git",
+    "url": "https://github.com/mavoweb/mavo.git"
+  },
+  "author": "Lea Verou",
+  "license": "MIT",
+  "bugs": {
+    "url": "https://github.com/mavoweb/mavo/issues"
+  },
+  "homepage": "https://mavo.io",
+  "autoupdate": {
```
Comment (xtuc) [inline ajax/libs/mavo/package.json]: this doesn't look good to me. `npm*` keys shouldn't be in the `autoupdate` key. But we should have a proper CI to enforce that

**Comment by xtuc** [comment]
@DmitrySharabin no worries, I will fix the format and add a CI

**Comment by DmitrySharabin** [comment]
@xtuc, thank you very much! I appreciate your help.

**Comment by xtuc** [comment]
fixed the configuration in https://github.com/cdnjs/cdnjs/commit/7ec780cbe665ae32b81f943ce022012d40efc9eb

**Pull request merged**